src/pyramid_debugtoolbar/panels/request_vars.py

```diff
@@ -103,9 +103,39 @@ def __init__(self, request):
                 lambda v: self.process_lazy_attr(attr_, is_dict, v),
             )
 
+        # safely displaying the POST information is a bit tedious
+        post_variables = None
+        post_body_info = {'size': len(request.body),
+                          'preview_bytes': None,
+                          }
+        try:
+            # under py2
+            #     iterating request.POST in webob-1.8.1 will trigger an
+            #     exception here which will kill the `response` object and break
+            #     at-least the headers panel
+            #     `curl -d "@/path/to/file.jpg" -X POST http://app`
+            # under py3:
+            #      the info seems decoded, but we can have 1400 items
+            post_variables = [(saferepr(k), saferepr(v)) for k, v in request.POST.items()]
+        except:
+            pass
+        if not post_variables and request.body:
+            # try to convert the post if it's text...
+            _post_preview = None
+            try:
+                try:
+                    _post_converted = request.body.decode('utf-8')
```

Comment:
I think you just want `request.text` here which will take into account the charset.

Reply:
yes. thanks!